# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JitActivation.h
# Commit: 1f21e8f610a7
# Full Hash: 1f21e8f610a72d6c6bd84321f41fb257335213e4
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-31 21:05:17
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
#   All these places require short fixes to use our new scheme for getting
# ==============================================================================

diff -r c50443ca5619 -r 1f21e8f610a7 js/src/vm/JitActivation.h
--- a/js/src/vm/JitActivation.h	Sat Oct 31 09:56:05 2020 +0000
+++ b/js/src/vm/JitActivation.h	Sat Oct 31 09:56:05 2020 +0000
@@ -214,6 +214,9 @@
     return reinterpret_cast<wasm::Frame*>(
         wasm::Frame::toJitEntryCaller(packedExitFP_));
   }
+  wasm::TlsData* wasmExitTls() const {
+    return wasm::GetNearestEffectiveTls(wasmExitFP());
+  }
   void setWasmExitFP(const wasm::Frame* fp) {
     if (fp) {
       MOZ_ASSERT(!wasm::Frame::isExitOrJitEntryFP(fp));